apps/web/lib/trpc/routers/keySettings.ts

```diff
@@ -0,0 +1,409 @@
+import { db, eq, schema } from "@/lib/db";
+import { TRPCError } from "@trpc/server";
+import { newId } from "@unkey/id";
+import { z } from "zod";
+import { auth, t } from "../trpc";
+
+export const keySettingsRouter = t.router({
+  updateEnabled: t.procedure
+    .use(auth)
+    .input(
+      z.object({
+        keyId: z.string(),
+        workspaceId: z.string(),
+        enabled: z.boolean(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const key = await db.query.keys.findFirst({
+        where: (table, { eq }) => eq(table.id, input.keyId),
+        with: {
+          workspace: true,
+        },
+      });
+      if (!key) {
+        throw new Error("key not found");
```

Comment:
inside tRPC routes, you should throw `new TRPCError({ code "", message: ""})`

that way it automatically handles the status code etc and sends along the `message` to the toaster